### PR TITLE
Bug 1833329: Add support for RemovePodsHavingTooManyRestarts strategy

### PR DESCRIPTION
**Pull request by damemi**
This adds the `RemovePodsHavingTooManyRestarts` strategy, added upstream in https://github.com/kubernetes-sigs/descheduler/pull/254 and synced to our fork in https://github.com/openshift/descheduler/pull/29.

This also updates the sample CR in the readme to show how to configure this strategy, and I made the change to now convert strategy parameter names to lowercase in our logic, allowing CamelCase params that are imo more readable.

### CODE REVIEW

File: pkg/operator/target_config_reconciler.go [hunk 1]
```diff
@@ -213,6 +214,29 @@ func generateConfigMapString(requestedStrategies []deschedulerv1beta1.Strategy)
 			}
 		case "nodetaints", "removepodsviolatingnodetaints":
 			policy.Strategies["RemovePodsViolatingNodeTaints"] = deschedulerapi.DeschedulerStrategy{Enabled: true}
+		case "removepodshavingtoomanyrestarts":
+			podsHavingTooManyRestarts := deschedulerapi.PodsHavingTooManyRestarts{}
```
Comment (ingvagabund) [inline pkg/operator/target_config_reconciler.go:218]: Also, let's wait until https://github.com/kubernetes-sigs/descheduler/pull/276 gets merged.

File: pkg/operator/target_config_reconciler.go [hunk 2]
```diff
@@ -177,7 +178,7 @@ func generateConfigMapString(requestedStrategies []deschedulerv1beta1.Strategy)
 				if err != nil {
 					return "", err
 				}
-				switch param.Name {
+				switch strings.ToLower(param.Name) {
```
[the remark this answers is not in the document]
Comment (damemi) [reply pkg/operator/target_config_reconciler.go:181]: I don't think that's really necessary, it's just one line (except for the Readme change) and functionally makes no difference to the existing behavior. But if it's easier for review I can if you'd like

File: pkg/operator/target_config_reconciler.go (hunk 1 above)
Comment (damemi) [reply pkg/operator/target_config_reconciler.go:218]: Agreed

File: pkg/operator/target_config_reconciler.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ingvagabund) [reply pkg/operator/target_config_reconciler.go:181]: There are other changes in README.md going from lower case to upper case. It's no big deal. It just makes clearer which change is adding the new strategy and which change is allowing the params names to be mix of lower and upper cases.

File: pkg/operator/target_config_reconciler.go (hunk 1 above)
Comment (damemi) [reply pkg/operator/target_config_reconciler.go:218]: @ingvagabund the upstream PR has merged, so I updated this as well. Should be all good for a final pass

**Comment by damemi** [comment]
/retitle Bug 1833329: Add support for RemovePodsHavingTooManyRestarts strategy

**Pull request merged**